compiler/ir/ir.psi2ir/src/org/jetbrains/kotlin/psi2ir/generators/DataClassMembersGenerator.kt

```diff
@@ -254,13 +256,17 @@ class DataClassMembersGenerator(
                     irLetS(
                         irGet(irPropertyType, receiver, getterSymbol)
                     ) { variable ->
+                        val get = irGet(irPropertyType, variable)
                         irIfNull(
                             context.irBuiltIns.intType,
                             irGet(irPropertyType, variable),
                             irInt(0),
                             getHashCodeOf(
                                 propertyType,
-                                irGet(irPropertyType, variable)
+                                if (irPropertyType.isMarkedNullable())
```

Comment:
Why is this implicit cast required?

Reply:
Good catch - it's not since there is an implicit coercion between the branches of a when expression and the result. This was from a separate branch where I experimented with removing the implicit coercions between IR elements. Removed.